client/extensions/woocommerce/app/settings/shipping/shipping-class-dialog.js

```diff
@@ -0,0 +1,213 @@
+				disabled: ! canSave,
+			},
+		];
+
+		if ( ! isNew ) {
+			buttons.unshift( {
+				action: 'delete',
+				label: (
+					<span>
+						<Gridicon icon="trash" /> { translate( 'Delete this class' ) }
+					</span>
+				),
+				onClick: this.onDelete,
+				additionalClassNames: 'shipping__class-delete is-scary is-borderless',
+			} );
+		}
+
+		const dialogProps = {
+			additionalClassNames: 'shipping__class-dialog woocommerce',
+			isVisible: true,
+			buttons: buttons,
+			onClose: this.onCancel,
+		};
+
+		return (
+			<Dialog { ...dialogProps }>
+				<div className="shipping__class-dialog-header">
+					<h3 className="shipping__class-dialog-heading">
+						{ isNew ? translate( 'Add shipping class' ) : translate( 'Edit shipping class' ) }
+					</h3>
+				</div>
+
+				<FormFieldSet className="shipping__class-dialog-name">
+					<FormLabel>{ translate( 'Name' ) }</FormLabel>
+					<FormTextInput type="text" name="name" value={ data.name } onChange={ this.onChange } />
+				</FormFieldSet>
+
+				<FormFieldSet className="shipping__class-dialog-slug">
+					<FormLabel>{ translate( 'Slug' ) + ': ' }</FormLabel>
+					<FormClickToEditInput
+						onChange={ this.onSlugChange }
+						value={ this.getSlugToDisplay() }
+						placeholder="-"
+						editAriaLabel={ translate( 'Edit slug' ) }
+						updateAriaLabel={ translate( 'Update slug' ) }
+					/>
+				</FormFieldSet>
+
+				<FormFieldSet>
+					<FormLabel>{ translate( 'Description' ) }</FormLabel>
+					<FormTextInput
+						type="text"
+						name="description"
+						value={ data.description }
+						onChange={ this.onChange }
+					/>
+				</FormFieldSet>
+			</Dialog>
+		);
+	}
+
+	getSlugToDisplay() {
+		const { name, slug } = this.props.data;
+
+		return slug.length ? slug : name.toLowerCase().replace( /\s+/g, '-' );
+	}
+
+	onChange = ( { target: { name, value } } ) => {
+		const { siteId, updateFieldValue } = this.props;
+
+		updateFieldValue( siteId, name, value );
+	};
+
+	onSlugChange = value => {
+		const { siteId, updateFieldValue } = this.props;
+
+		updateFieldValue( siteId, 'slug', value );
+	};
+
+	onCancel = () => {
+		const { close, siteId } = this.props;
+
+		close( siteId );
+	};
+
+	onDelete = () => {
+		const {
+			data: { id },
+			siteId,
+			deleteClass,
+		} = this.props;
+
+		deleteClass( siteId, id );
+	};
+
+	onSave = () => {
+		const {
+			existingSlugs,
+			siteId,
+			translate,
+			data: { id, name, slug },
+			updateFieldValue,
+			save,
+		} = this.props;
+
+		const slugToUse = this.getSlugToDisplay();
+
+		// Check if the slug is free
+		if ( -1 !== existingSlugs.indexOf( slugToUse ) ) {
+			const text = translate(
+				'There was a problem saving %(name)s. A shipping class with this slug already exists.',
+				{
+					args: { name },
+				}
+			);
+
+			return this.props.errorNotice( text, {
+				id: `shipping-class-error-${ id }`,
+			} );
+		}
+
+		// If there is no slug, use the generated one
+		if ( 0 === slug.trim().length ) {
```

Comment:
Is there a reason `trim` is used here but not in the `getSlugToDisplay` function? May not matter much but it seems like that would allow some inputs (like ` `) to fly under the radar.

Reply:
Good point, the slug is now always trimmed.